src/extension/tags.c

```diff
@@ -843,20 +976,43 @@ static PHP_FUNCTION(datadog_appsec_track_user_login_failure_event)
     }
     zend_array *meta_ht = Z_ARRVAL_P(meta);
 
-    // appsec.events.users.login.failure.usr.id = <user_id>
-    _add_custom_event_keyval(
-        meta_ht, _dd_login_failure_event, _dd_tag_user_id, user_id, true);
+    if (user_id != NULL && ZSTR_LEN(user_id) > 0) {
+        // appsec.events.users.login.failure.usr.id = <user_id>
+        _add_custom_event_keyval(meta_ht, _dd_login_failure_event,
+            _dd_tag_user_id, user_id, true, !automated);
+    }
 
     // appsec.events.users.login.failure.track = true
-    _add_custom_event_keyval(
-        meta_ht, _dd_login_failure_event, _track_zstr, _true_zstr, true);
+    _add_custom_event_keyval(meta_ht, _dd_login_failure_event, _track_zstr,
+        _true_zstr, true, !automated);
+
+    // manual.keep = true
+    _add_new_zstr_to_meta(
+        meta_ht, _dd_tag_manual_keep_zstr, _true_zstr, true, !automated);
```

Comment:
As far as I'm aware this is equivalent to setting the sampling priority, which is already done at the end of the function, can you double check? If it is, you can remove it.

Reply:
`manual.keep` is a flag to tell the tracer to send the trace and ignore the sampler, however this is already done with the `dd_tags_set_sampling_priority()` function, although in a different way. So you don't need to add `manual.keep` at all.

Reply:
ok